AirCasting/Models/AirBeamCellularSessionCreator.swift

```diff
@@ -0,0 +1,95 @@
+// Created by Lunar on 13/09/2021.
+//
+
+import Foundation
+import CoreLocation
+
+final class AirBeamCellularSessionCreator: SessionCreator {
+    enum AirBeamSessionCreatorError: Swift.Error {
+        case invalidCreateSessionContext(CreateSessionContext)
+    }
+    let userAuthenticationSession: UserAuthenticationSession
+    let measurementStreamStorage: MeasurementStreamStorage
+    private let createSessionService: CreateSessionAPIService
+    
+    convenience init(measurementStreamStorage: MeasurementStreamStorage, userAuthenticationSession: UserAuthenticationSession, baseUrl: BaseURLProvider) {
+        self.init(measurementStreamStorage: measurementStreamStorage,
+                  createSessionService: CreateSessionAPIService(authorisationService: userAuthenticationSession, baseUrlProvider: baseUrl),
```

Comment:
I don't quite like this introduction of low-level detail which is a `CreateSessionAPIService` here just for convenience reasons. I think we should remove this convenience init and only use interfaces. Moving purely clean-code reasons aside, this kind of import will cause longer compile times. This is because each time something inside the file containing `CreateSessionAPIService` changes it will also force this file to recompile (and all stuff that depends on it too, etc. etc.). I've seen iOS projects with 1h+ compile times. Not fun at all. This is why the [dependency inversion principle](https://stackify.com/dependency-inversion-principle/) is really important to follow